### PR TITLE
Explain how to compile the TypeScript code sample

### PR DESCRIPTION
**Pull request by PierreFritsch**
* Clarify how to implement the recommendation given on the capire page [Using TypeScript](https://cap.cloud.sap/docs/get-started/using-typescript):

    > For productive usage always precompile TypeScript code to JavaScript due to performance reasons and use the cds binary.

* Add `tsconfig.json` in order to ensure that the generated JavaScript files can be understood by CAP and prevent error messages like: 
    > Service "say" has no handler for "hello".

### CODE REVIEW

**Comment by PierreFritsch** [comment]
Build failing due to `package-lock.json` being out of sync.

**Comment by renejeglinsky** [comment]
PING :) @danjoa @chgeo

**Comment by PierreFritsch** [comment]
Originally proposed to:

* Extract the Hello World! TypeScript code sample to a dedicated folder in order to avoid polluting the Hello World! JavaScript code sample.

but the CAP team decided not have a separate variant of the `hello` package (communicated internally by @chgeo), so moved back the missing TypeScript pieces to this folder.

**Comment by PierreFritsch** [comment]
Internal ticket is cap/issues#9166. 
This PR was adapted with the requested changes.

File: hello/srv/world.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import type { Request } from "@sap/cds/apis/services"
```
Comment (manwithafox) [inline hello/srv/world.ts:1]: Looks like this file is replaced by the file with the same name in `hello/src/srv` folder.
Can this version be removed?

**Comment by PierreFritsch** [comment]
Closing unmerged. Conflicts in the files: 

```
hello/README.md
hello/package.json
hello/srv/world.ts
hello/test/hello-world-ts.test.ts
package-lock.json
package.json
```

**Pull request closed without merging**